Лабораторная работа 1

lin-shl opened the pull request:
![выполненные_задания](https://github.com/user-attachments/assets/8f1d9f4d-3040-470b-a8d5-381f90c033a9)

Во 2 задании поначалу возникли сложности с логическим пониманием, что на что и сколько раз нужно умножить, поэтому код длинный и переполнен комментариями. После уже стало понятно, что можно было просто всё перемножить, да и в peek solution тоже было написано, но код был уже красиво оформлен, я не стала его менять :)

Задание 3 было уже выполнено в курсе, поэтому я его записала немного другим способом.

lin-shl committed:
Add files via upload

TimurIslamgulov commented:
Красивое оформление - наше всё )

TimurIslamgulov commented on Лабораторная_работа_1/task_1.py:
```diff
@@ -0,0 +1,5 @@
+numbers = [2, -93, -2, 8, None, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+
+numbers[4] = (sum((sum(numbers[:4]), sum(numbers[5:]))) / len(numbers))
```
откуда взялась 4? надо найти кодом, то есть вам нужна новая переменная

lin-shl replied:
Просто поскольку в этой лабораторной не проходились никакие операторы типа условий или циклов, я подумала, что нужно найти через глаза х) Тогда вопрос: я должна найти это через циклы/условия или я просмотрела какой-то ещё оператор, который мне этот None позволит без перебора массива выявить?

TimurIslamgulov commented on Лабораторная_работа_1/task_2.py:
```diff
@@ -0,0 +1,14 @@
+diskette = 1.44 # объем дискеты в Мб
+diskette_in_bite = diskette * pow(1024, 2) # объем дискеты в байтах
+
+amount = int(diskette_in_bite // b_weight) # кол-во книг на дискете
```
а зачем вам и int и //? масло масляное получается

lin-shl replied:
Мне показалось логичным, что тут не целочисленный тип надо использовать, а деление без остатка, потому что нам нужно найти именно сколько книг поместится, то есть только книги без лишних страничек, значит // деление. Но тогда даёт ошибку в Check 
<img width="476" alt="Снимок экрана 2024-10-20 в 21 28 32" src="https://github.com/user-attachments/assets/940161c4-b414-460a-9a0a-0a578c2f3110">

TimurIslamgulov replied:
так просто int бы хватило

TimurIslamgulov reviewed (commented):
2 замечания

lin-shl committed:
Update task_2.py

Изменила деление в amount на деление с остатком. Заменила названия переменных, чтобы они были понятнее. Доработала код в соответствии с PEP8 (логические пустые строки, комментарии с большой буквы).

lin-shl committed:
Update task_1.py

Упустила из виду метод index.

TimurIslamgulov commented on Лабораторная_работа_1/task_1.py:
```diff
@@ -0,0 +1,6 @@
+numbers = [2, -93, -2, 8, None, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+
+wrong_index = numbers.index(None)
+numbers[wrong_index] = (sum((sum(numbers[:4]), sum(numbers[5:]))) / len(numbers))
```
опять магические цифры, почему так?

lin-shl replied:
блин, я забыла поменять значения в суммах... =(

lin-shl committed:
Update task_1.py

Ну теперь-то точно ничего не забыла! XD
